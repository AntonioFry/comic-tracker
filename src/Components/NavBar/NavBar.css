.nav-bar {
  animation-duration: 1s;
  animation-name: slideout-nav;
  align-content: flex-start;
  background-color: black;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  left: -25%;
  padding: 20px 0px;
  position: absolute;
  width: 25%;
  height: 100%;
}

.nav-bar-start {
  align-content: flex-start;
  background-color: black;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  left: -25%;
  padding: 20px 0px;
  position: absolute;
  width: 25%;
  height: 100%;
}

.nav-bar-toggled {
  animation-duration: 1s;
  animation-name: slidein-nav;
  align-content: flex-start;
  background-color: black;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  padding: 20px 0px;
  position: absolute;
  width: 25%;
  height: 100%;
}

.nav-bar-links {
  color: white;
  font-size: 20px;
  margin: 10px 0px;
  padding: 10px;
  text-align: center;
  text-decoration-line: none;
  width: 100%;
}

.nav-bar-links:hover {
  background: white;
  color: black;
}

.icon {
  animation-duration: 1s;
  animation-name: slideout-icon;
  background-color: black;
  width: 60px;
  position: fixed;
  left: 0%;
}

.icon-start {
  background-color: black;
  width: 60px;
  position: fixed;
  left: 0%;
}

.icon-toggled {
  animation-duration: 1s;
  animation-name: slidein-icon;
  background-color: black;
  width: 60px;
  position: fixed;
  left: 25%;
}

@keyframes slidein-nav {
  from {
    left: -25%;
  }
  to {
    left: 0%;
  }
}

@keyframes slideout-nav {
  from {
    left: 0%;
  }
  to {
    left: -25%;
  }
}

@keyframes slidein-icon {
  from {
    left: 0%;
  }
  to {
    left: 25%;
  }
}

@keyframes slideout-icon {
  from {
    left: 25%;
  }
  to {
    left: 0%;
  }
}