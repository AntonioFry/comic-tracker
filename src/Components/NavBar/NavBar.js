import React from 'react';
import './NavBar.css';
import { Link } from 'react-router-dom';

export const NavBar = ({ toggled, toggleNavBar }) => {
  let iconToggled;
  let navBarToggled;
  let backgroundContainer;
  if (toggled === null) {
    navBarToggled = 'nav-bar-start';
    iconToggled = 'icon-start';
  } else {
    iconToggled = toggled === true ? 'icon-toggled' : 'icon';
    navBarToggled = toggled === true ? 'nav-bar-toggled' : 'nav-bar';
    backgroundContainer = toggled === true ? 'background-toggled' : 'background-div';
  }
  return (
    <div class={backgroundContainer}>
      <nav className={navBarToggled}>
        <img className={iconToggled} src={require('../../../public/hamburger-icon.png')} onClick={() => toggleNavBar()}/>
        <Link className="nav-bar-links" to="/">HOME</Link>
        <Link className="nav-bar-links" to="/character">CHARACTERS</Link>
        <Link className="nav-bar-links" to="/saved-comics">SAVED</Link>
      </nav>
    </div>
  )
}