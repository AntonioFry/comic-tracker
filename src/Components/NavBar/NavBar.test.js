import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom';
import { NavBar } from './NavBar.js';
import { BrowserRouter } from 'react-router-dom';

describe('NavBar', () => {

  it('Should match the snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});