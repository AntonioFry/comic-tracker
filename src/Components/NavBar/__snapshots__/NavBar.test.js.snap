exports[`NavBar Should match the snapshot 1`] = `
<div
  className="background-div">
  <nav
    className="nav-bar">
    <img
      className="icon"
      onClick={[Function]}
      src="hamburger-icon.png" />
    <a
      className="nav-bar-links"
      href="/"
      onClick={[Function]}>
      HOME
    </a>
    <a
      className="nav-bar-links"
      href="/characters"
      onClick={[Function]}>
      CHARACTERS
    </a>
    <a
      className="nav-bar-links"
      href="/saved-comics"
      onClick={[Function]}>
      SAVED
    </a>
  </nav>
</div>
`;
